Core/Object Arts/Dolphin/MVP/MultiSelectListViewTest.cls

```diff
@@ -17,6 +17,116 @@ initializePresenter
 sortSelections
 	^#(49 50)!
 
+testNewSelectionsClickOutsideListWithModifiers
+	| event |
+	self setUpForSelectionTesting.
+	presenter selectionsByIndex: #(1).
+	(OrderedCollection new)
+		add: #(#control);
+		add: #(#shift);
+		add: #(#control #shift);
+		do: 
+				[:modifiers |
+				event := self mouseDownEventOnItem: 0 buttons: (modifiers copyWith: #left).
+				self assert: (presenter newSelectionsFromEvent: event) = #(1)].!
+
+testNewSelectionsCtrlClickOnSelectedItem
+	| event |
+	self setUpForSelectionTesting.
+	presenter selectionsByIndex: #(1 2).
+	event := self mouseDownEventOnItem: 1 buttons: #(#left #control).
+	self assert: (presenter newSelectionsFromEvent: event) = #(2).
+	presenter selectionsByIndex: #(1).
+	self assert: (presenter newSelectionsFromEvent: event) = #().!
+
+testNewSelectionsCtrlClickOnUnselectedItem
+	| event |
+	self setUpForSelectionTesting.
+	event := self mouseDownEventOnItem: 1 buttons: #(#left #control).
+	self assert: (presenter newSelectionsFromEvent: event) = #(1).
+	presenter selectionsByIndex: #(2).
+	self assert: (presenter newSelectionsFromEvent: event) = #(1 2).!
+
+testNewSelectionsCtrlShiftClickAdditive
+	| event |
+	self setUpForSelectionTesting.
+	presenter
+		selectionsByIndex: #(1 3);
+		selectionMarkIndex: 3.
+	event := self mouseDownEventOnItem: 7 buttons: #(#left #control #shift).
+	self assert: (presenter newSelectionsFromEvent: event) = #(1 3 4 5 6 7).
+	presenter selectionsByIndex: #(1 3 5 7).
+	self assert: (presenter newSelectionsFromEvent: event) = #(1 3 4 5 6 7).!
+
+testNewSelectionsCtrlShiftClickSubtractive
+	| event |
+	self setUpForSelectionTesting.
+	presenter
+		selectionsByIndex: #(1 2 3 4 5 7);
+		selectionMarkIndex: 6.
+	event := self mouseDownEventOnItem: 2 buttons: #(#left #control #shift).
+	"In Windows Explorer and other applications that use ListView or a similar control,
```

Comment:
As mentioned earlier, probably a separate bug that we ought to fix. That will be easier to do without breaking other selection behaviour now you have put in place these tests.